cli/azd/pkg/project/project_config.go

```diff
@@ -99,30 +98,25 @@ func (pc *ProjectConfig) GetProject(ctx *context.Context, env *environment.Envir
 		*ctx = telemetry.ContextWithTemplate(*ctx, project.Metadata.Template)
 	}
 
-	if pc.ResourceGroupName == "" {
-		// We won't have a ResourceGroupName yet if it hasn't been set in either azure.yaml or AZURE_RESOURCE_GROUP env var
-		// Let's try to find the right resource group for this environment
-		resourceManager := infra.NewAzureResourceManager(*ctx)
-		resourceGroupName, err := resourceManager.FindResourceGroupForEnvironment(*ctx, env)
-		if err != nil {
-			return nil, err
-		}
-		pc.ResourceGroupName = resourceGroupName
+	resourceGroupName, err := GetResourceGroupName(*ctx, pc, env)
```

Comment:
There isn't anything that was using `projectConfig.ResourceGroupName` which is why this is safe to refactor to `project.ResourceGroupName`